pkg/controllers/edgeconnect/controller.go

```diff
@@ -144,49 +164,74 @@ func (controller *Controller) reconcileEdgeConnectDeletion(ctx context.Context,
 }
 
 func (controller *Controller) reconcileEdgeConnect(ctx context.Context, edgeConnect *edgeconnectv1alpha1.EdgeConnect) (reconcile.Result, error) {
+	llog := log.WithValues("namespace", edgeConnect.Namespace, "name", edgeConnect.Name)
+
+	logExit := llog.DebugLogFunctionBoundaries()
+	defer logExit()
+
 	oldStatus := *edgeConnect.Status.DeepCopy()
 
 	err := controller.reconcileEdgeConnectCR(ctx, edgeConnect)
 	if err != nil {
 		edgeConnect.Status.SetPhase(status.Error)
-		log.Error(err, "error reconciling EdgeConnect", "namespace", edgeConnect.Namespace, "name", edgeConnect.Name)
+		llog.Debug("error reconciling EdgeConnect, setting phase 'Error'")
 	} else {
+		llog.Debug("moving EdgeConnect to phase 'Running'")
 		edgeConnect.Status.SetPhase(status.Running)
 	}
 
 	if isDifferentStatus, err := hasher.IsDifferent(oldStatus, edgeConnect.Status); err != nil {
-		log.Error(errors.WithStack(err), "failed to generate hash for the status section")
+		llog.Error(errors.WithStack(err), "failed to generate hash for the status section")
 	} else if isDifferentStatus {
-		log.Info("status changed, updating EdgeConnect")
+		llog.Info("status changed, updating EdgeConnect")
 
 		if errClient := controller.updateEdgeConnectStatus(ctx, edgeConnect); errClient != nil {
-			return reconcile.Result{RequeueAfter: fastUpdateInterval}, errors.WithMessagef(errClient, "failed to update EdgeConnect after failure, original error: %s", err)
+			retErr := errors.WithMessagef(errClient, "failed to update EdgeConnect after failure, original error: %s", err)
+
+			llog.Debug("reconcileEdgeConnect error")
+
+			return reconcile.Result{RequeueAfter: fastUpdateInterval}, retErr
 		}
 	}
 
-	log.Info("reconciling EdgeConnect done", "name", edgeConnect.Name, "namespace", edgeConnect.Namespace)
+	llog.Info("reconciling EdgeConnect done")
 
 	if err != nil {
+		llog.Debug("reconcileEdgeConnect error")
+
```

Comment:
feels a bit redundant when you already have:
```
	logExit := llog.DebugLogFunctionBoundaries()
	defer logExit()
```

Reply:
That's food for thought, the dedicated Exit log gives better information, the automated one has kind of the same information about the exit point but more hidden in a filename:linenumber value added to the log.

Reply:
fixed in e3699102